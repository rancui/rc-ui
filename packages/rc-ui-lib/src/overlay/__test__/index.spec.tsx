import React from 'react';
import { render } from '@testing-library/react';
import { Overlay } from '..';

describe('Overlay', () => {
  it('should change z-index when using z-index prop', () => {
    render(<Overlay visible zIndex={99} />);
    const overlay = document.querySelector('.rc-overlay');
    expect(getComputedStyle(overlay).zIndex).toEqual('99');
  });

  it('should allow to custom class name with class-name prop', () => {
    render(<Overlay visible className="foo" />);
    const overlay = document.querySelector('.rc-overlay');
    expect(overlay.classList.contains('foo')).toBeTruthy();
  });

  it('should allow to custom style with custom-style prop', () => {
    render(<Overlay visible customStyle={{ backgroundColor: 'red' }} />);
    const overlay = document.querySelector('.rc-overlay');
    expect(getComputedStyle(overlay).backgroundColor).toEqual('red');
  });

  it('should change animation duration when using duration prop', () => {
    render(<Overlay visible duration={100} />);
    const overlay = document.querySelector('.rc-overlay');
    expect(getComputedStyle(overlay).animationDuration).toEqual('100ms');
  });

  it('should render default slot correctly', () => {
    render(<Overlay visible>Custom Default</Overlay>);
    expect(document.querySelector('.rc-overlay')).toMatchSnapshot();
  });
});
