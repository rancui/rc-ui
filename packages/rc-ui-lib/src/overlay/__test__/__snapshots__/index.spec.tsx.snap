// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Overlay should render default slot correctly 1`] = `
<div
  class="rc-overlay"
  style="animation-duration: 300ms;"
>
  Custom Default
</div>
`;
